### PR TITLE
Updated XNNPACK to v0.0.20210622

### PR DESCRIPTION
**Pull request by stemann**
Dependency for #4554

### CODE REVIEW

File: X/XNNPACK/build_tarballs.jl [hunk 1]
```diff
@@ -74,17 +71,22 @@ install_license ../LICENSE
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = supported_platforms()
-filter!(p -> !(Sys.iswindows(p) && arch(p) == "i686"), platforms) # i686-Windows fails to link
+filter!(p -> arch(p) != "armv6l", platforms) # armv6l is unsupported by XNNPACK (lacks NEON instructions)
 filter!(p -> arch(p) != "powerpc64le", platforms) # PowerPC64LE is unsupported by XNNPACK (Unsupported architecture in src/init.c)
+platforms = expand_microarchitectures(platforms, [
```
Comment (giordano) [inline X/XNNPACK/build_tarballs.jl:76]: Please don't use it, it doesn't work.

**Review by giordano — CHANGES_REQUESTED** [review]

File: X/XNNPACK/build_tarballs.jl (hunk 1 above)
Comment (stemann) [reply X/XNNPACK/build_tarballs.jl:76]: OK :-)

Comment (stemann) [reply X/XNNPACK/build_tarballs.jl]: Ah - it actually enables the armv7l build (for neonvfpv4). Is that acceptable? (just one micro-architecture, for a non-tier-1 platform)

Comment (giordano) [reply X/XNNPACK/build_tarballs.jl]: The whole microarchitecture stuff doesn't work. Specifically, the artifact selection in Pkg is broken. There is a reason why there isn't a single builder using it

Comment (stemann) [reply X/XNNPACK/build_tarballs.jl]: OK - removing it.

File: X/XNNPACK/bundled/patches/xnnpack-soversion.patch [hunk 2]
```diff
@@ -12,4 +12,4 @@ index 9bcaa5816..dbe149f85 100644
 +  SOVERSION 0)
  IF(CMAKE_SYSTEM_PROCESSOR MATCHES "^armv[5-8]" OR IOS_ARCH MATCHES "^armv7")
    SET_PROPERTY(SOURCE ${XNNPACK_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -marm ")
-   SET_PROPERTY(SOURCE ${XNNPACK_PSIMD_FASTMATH_MICROKERNEL_SRCS} ${XNNPACK_PSIMD_ACCMATH_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -mfpu=neon ")
+   SET_PROPERTY(SOURCE ${XNNPACK_NEON_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -march=armv7-a -mfpu=neon ")
```
Comment (giordano) [inline X/XNNPACK/bundled/patches/xnnpack-soversion.patch:15]: Why is this necessary?

Comment (stemann) [reply X/XNNPACK/bundled/patches/xnnpack-soversion.patch:15]: I just updated the patches, so they (should apply) applied more nicely.

Comment (stemann) [reply X/XNNPACK/bundled/patches/xnnpack-soversion.patch:15]: PSIMD was removed as a dependency of XNNPACK.

Comment (giordano) [reply X/XNNPACK/bundled/patches/xnnpack-soversion.patch:15]: I was more specifically referring to `-march=armv7-a`.

Comment (stemann) [reply X/XNNPACK/bundled/patches/xnnpack-soversion.patch:15]: Those lines are just the context of the patch: https://github.com/google/XNNPACK/blob/79cd5f9e18ad0925ac9a050b00ea5a36230072db/CMakeLists.txt#L3473

I was just trying to edit as little as possible in the patches, so this one only inserts a VERSION and SOVERSION: https://github.com/google/XNNPACK/blob/79cd5f9e18ad0925ac9a050b00ea5a36230072db/CMakeLists.txt#L3470

Comment (stemann) [reply X/XNNPACK/bundled/patches/xnnpack-soversion.patch:15]: ... the `march` arguments is being removed by https://github.com/JuliaPackaging/Yggdrasil/pull/5416/files#diff-8eb5c3a4ae80bba315cf237677664f0c81196496be3bf80c10a394d64c9e5563

Comment (giordano) [reply X/XNNPACK/bundled/patches/xnnpack-soversion.patch:15]: Reading the diff of a diff is hard :weary:

Comment (stemann) [reply X/XNNPACK/bundled/patches/xnnpack-soversion.patch:15]: Indeed :-)

Upstreaming patches...

**Review by giordano — APPROVED** [review]
